### PR TITLE
fix: combining splunk versions for python3.9 matrix issues

### PR DESCRIPTION
**Pull request by pbajaj-crest**
## Summary 
- Update reusable workflow according to updated handling of python 3.9 tests

### Issue ticket number and link :- [Jira Ticket](https://splunk.atlassian.net/browse/ADDON-69651)
- Job in AWS addon where all the relevent  test is running successfully with python3.9: [Job](https://github.com/splunk/splunk-add-on-for-amazon-web-services/actions/runs/8687485637/job/23821632227)

- Addon Job for MSCS :- [Job](https://github.com/splunk/splunk-add-on-for-microsoft-cloud-services/actions/runs/8683988655)

### Description 
- Updated workflow so that steps combined_Splunkmatrix and python39_splunk in job meta are merged into single step.
- Updated workflow so these tests are being executed the same way for all the other test types

### CODE REVIEW

**Review by mkolasinski-splunk — COMMENTED** [review]
Generally looks good.
Could you re-run tests with markers uncommented?

File: .github/workflows/branch-test-release.yml [hunk 1]
```diff
@@ -16,9 +16,9 @@ permissions:
 
 jobs:
   run-template-rollout:
-    uses: splunk/addonfactory-repository-template/.github/workflows/sync.yaml@${{ inputs.branchname }}
```
[the remark this answers is not in the document]
Comment (pbajaj-crest) [reply .github/workflows/branch-test-release.yml:19]: > Why is this file edited?

it was giving me error for white spacing thats why i have to edit that.

**Comment by pbajaj-crest** [comment]
> Generally looks good. Could you re-run tests with markers uncommented?

-sure i have re-run the test with markers.[Link](https://github.com/splunk/splunk-add-on-for-amazon-web-services/pull/1216/commits/6ce5a61591fbd0bec7ba4b48eef7be77647805d9) and  here the job details for that [job](https://github.com/splunk/splunk-add-on-for-amazon-web-services/actions/runs/8689960373)

**Comment by mkolasinski-splunk** [comment]
@pbajaj-crest Please rename PR title to follow semantic convention.

**Comment by pbajaj-crest** [comment]
> @pbajaj-crest Please rename PR title so it follows semantic convention.



> @pbajaj-crest Please rename PR title so it follows semantic convention.

Its done now.

**Review by mkolasinski-splunk — APPROVED** [review]

**Review by mkolasinski-splunk — APPROVED** [review]

**Pull request merged**